### PR TITLE
Add more descriptive error messages on password reset fail

### PR DESCRIPTION
**Pull request by PeterBeyond**
This was slightly more 'tricky' than I anticipated, as we get back an IEnumerable of errors if the reset password operation fails. I've taken the approach of grabbing the first error, and converting it to an enum so that we can maintain a consistent contract (like how we handle sign in).

I also removed a lot of the validation code in ResetPasswordHandler as it seemed to me like the _userManager.ResetPasswordAsync method would handle these cases.

### CODE REVIEW

File: src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs [hunk 1]
```diff
@@ -104,33 +104,25 @@ public Task SignOut(CancellationToken cancellationToken)
     }
 
     [HttpPost("ResetPassword")]
-    public async Task ResetPassword(
-        [FromBody] ResetPasswordRequest resetPassword, 
+    public async Task<ResetPasswordStatus> ResetPassword(
```
Comment (andriikaplanovskyi) [inline src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs:107]: @PeterBeyond, async / await no longer needed here:

```
public Task<ResetPasswordStatus> ResetPassword(...) 
  => return _mediator.Send(new ResetPassword(...),
           cancellationToken);
```

File: src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs [hunk 2]
```diff
@@ -104,33 +104,25 @@ public Task SignOut(CancellationToken cancellationToken)
...
-            await _mediator.Send(new ResetPassword(
-                   resetPassword.UserId,
-                   resetPassword.Token,
-                   resetPassword.Password),
-               cancellationToken).ConfigureAwait(false);
-        }
-        catch (Exception ex)
-        {
-            // Don't throw exception so we don't reveal which emails are currently in use
-            Logger.LogError("Failed to reset user password for user: {exception}", ex.Message);
```
Comment (andriikaplanovskyi) [inline src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs:122]: @PeterBeyond, do I understand right, we no longer log any ResetPassword errors (and their reasons), we just return the error code?

**Review by andriikaplanovskyi — CHANGES_REQUESTED** [review]

File: src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs (hunk 1 above)
Comment (PeterBeyond) [reply src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs]: Fixed

**Review by andriikaplanovskyi — APPROVED** [review]

File: src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs (hunk 2 above)
Comment (PeterBeyond) [reply src/OneBeyond.Studio.Obelisk.WebApi/Controllers/AuthController.cs:128]: Correct, as we are no longer throwing exceptions. I've moved the logging into the Handler now, to be consistent with how SignInHandler does things - thanks for raising this

**Pull request merged**